core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -462,7 +462,16 @@ private URL findApplicationResource(String name) {
         Path resourcePath = null;
 
         for (Path i : applicationClassDirectories) {
-            resourcePath = i.resolve(name);
+            // Resource names are always separated by the "/" character.
+            // Here we are trying to resolve those resources using a filesystem
+            // Path, so we replace the "/" character with the filesystem
+            // specific separator before resolving
+            String path = name;
+            final String pathSeparator = i.getFileSystem().getSeparator();
```

Comment:
This is a small nit but if you do this instead:

```java
if (File.separatorChar != '/') {
    path = name.replace('/', File.separatorChar);
}
```
then the code is eliminated on *NIX-like platforms.

Reply:
Hello David, I'll actually close this PR. I had opened this to help with https://github.com/quarkusio/quarkus/pull/5793. @mkouba has already pulled this commit into that PR and even incorporated your review comment into that change.